src/fastoad_cs25/models/geometry/geom_components/wing/components/compute_center_chord.py

```diff
@@ -0,0 +1,93 @@
+        self.add_input("data:geometry:wing:root:y", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:kink:chord", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:kink:leading_edge:x:local", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:kink:y", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:tip:chord", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:tip:leading_edge:x:local", val=np.nan, units="m")
+        self.add_input("data:geometry:wing:tip:y", val=np.nan, units="m")
+
+        self.add_output("data:geometry:wing:center:chord", units="m")
+        self.add_output("data:geometry:wing:center:leading_edge:x:local", units="m")
+
+    def setup_partials(self):
+        self.declare_partials(
+            "data:geometry:wing:center:chord",
+            [
+                "data:geometry:wing:root:chord",
+                "data:geometry:wing:root:y",
+                "data:geometry:wing:kink:chord",
+                "data:geometry:wing:kink:leading_edge:x:local",
+                "data:geometry:wing:kink:y",
+                "data:geometry:wing:tip:chord",
+                "data:geometry:wing:tip:leading_edge:x:local",
+                "data:geometry:wing:tip:y",
+            ],
+            method="fd",
+        )
+        self.declare_partials(
+            "data:geometry:wing:center:leading_edge:x:local",
+            [
+                "data:geometry:wing:root:chord",
+                "data:geometry:wing:root:y",
+                "data:geometry:wing:kink:chord",
+                "data:geometry:wing:kink:leading_edge:x:local",
+                "data:geometry:wing:kink:y",
+                "data:geometry:wing:tip:chord",
+                "data:geometry:wing:tip:leading_edge:x:local",
+                "data:geometry:wing:tip:y",
+            ],
+            method="fd",
+        )
+
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+        l2_wing = inputs["data:geometry:wing:root:chord"]
+        y2_wing = inputs["data:geometry:wing:root:y"]
+        l3_wing = inputs["data:geometry:wing:kink:chord"]
+        x3_wing = inputs["data:geometry:wing:kink:leading_edge:x:local"]
+        y3_wing = inputs["data:geometry:wing:kink:y"]
+        l4_wing = inputs["data:geometry:wing:tip:chord"]
+        x4_wing = inputs["data:geometry:wing:tip:leading_edge:x:local"]
+        y4_wing = inputs["data:geometry:wing:tip:y"]
+
+        if y3_wing <= y2_wing:
+            y_values = np.squeeze([y2_wing, y4_wing])
+            x_values = np.squeeze([[0.0], x4_wing])
+            l_values = np.squeeze([l2_wing, l4_wing])
+        else:
+            y_values = np.squeeze([y2_wing, y3_wing, y4_wing])
+            x_values = np.squeeze([[0.0], x3_wing, x4_wing])
+            l_values = np.squeeze([l2_wing, l3_wing, l4_wing])
+        x_interp = interp1d(y_values, x_values, fill_value="extrapolate")
```

Comment:
Out of the recommend substitutes for interp1d, two options are available: the use of numpy `interp` or to use splines. However, numpy's `interp` doesn't allow for extrapolation which we are clearly using here so my recommendation is to use the following

```suggestion
        x_interp = make_interp_spline(y_values, x_values, k=1)
```

But other options available are detailed here: https://docs.scipy.org/doc/scipy/tutorial/interpolate/1D.html#tutorial-interpolate-1dsection